omocha-infra/src/main/java/org/omocha/infra/mail/MailSenderImpl.java

```diff
@@ -0,0 +1,45 @@
+package org.omocha.infra.mail;
+
+import org.omocha.domain.mail.MailCommand;
+import org.omocha.domain.mail.MailSender;
+import org.omocha.domain.mail.exception.MailSendFailException;
+import org.omocha.domain.member.vo.Email;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.stereotype.Component;
+
+import jakarta.mail.MessagingException;
+import jakarta.mail.internet.MimeMessage;
+import lombok.RequiredArgsConstructor;
+
+@Component
+@RequiredArgsConstructor
+public class MailSenderImpl implements MailSender {
+
+	private final JavaMailSender mailSender;
+
+	@Override
+	public void sendMail(MailCommand.SendMail sendCommand, String code, String hostEmail) {
+
+		MimeMessage mail = createMimeMessage(sendCommand.email(), code, hostEmail);
+		mailSender.send(mail);
+	}
+
+	private MimeMessage createMimeMessage(Email email, String code, String fromEmail) {
+		MimeMessage mimeMessage = mailSender.createMimeMessage();
+
+		try {
+			mimeMessage.setFrom(fromEmail);
+			mimeMessage.setRecipients(MimeMessage.RecipientType.TO, email.getValue());
+			mimeMessage.setSubject("이메일 인증");
+			String body = "";
+			body += "<h3>" + "요청하신 인증 번호입니다." + "</h3>";
+			body += "<h1>" + code + "</h1>";
+			body += "<h3>" + "감사합니다." + "</h3>";
+			mimeMessage.setText(body, "UTF-8", "html");
+		} catch (MessagingException e) {
+			throw new MailSendFailException(email.getValue());
+		}
+
+		return mimeMessage;
+	}
+}
```

Comment:
createMimeMessage는 domain 계층으로 올라가는게 좋아보입니다!
어떤 메세지를 만들지는 infra의 관심사가 아닌 service의 관심사라고 생각되요.

또 위에 코멘트 남긴거대로 fromEmail이 infra로 넘어오면 해당 파라미터는 사라지게 될 거고
아까 저랑 말하면서 말씀하신 createMimeMessage를 유동적으로 사용하고 싶다 하신 걸 고려해보면
```suggestion
	private MimeMessage createMimeMessage(Email email, String code, String subject, String body) {
		MimeMessage mimeMessage = mailSender.createMimeMessage();

		try {
			mimeMessage.setFrom(fromEmail);
			mimeMessage.setRecipients(MimeMessage.RecipientType.TO, email.getValue());
			mimeMessage.setSubject(subject);
			mimeMessage.setText(body, "UTF-8", "html");
		} catch (MessagingException e) {
			throw new MailSendFailException(email.getValue());
		}

		return mimeMessage;
	}
}
```
이런식으로 코드가 변경될 수 있겠네요.
subject랑 body는 해당 메서드를 호출하는 쪽에서 만들어서 넘겨주면 될 것 같구요.

근데 좀 생각해보니 이건 MimeMessage를 어디에 두냐에 따라 좀 달라질 것 같네요.
현재는 JavaMailSender가 MimeMessage를 사용하는데 만약 다른 라이브러리가 MimeMessage를 사용하지 않는다면
email 전송 라이브러리가 변경될 때 MimeMessage도 다른것으로 대체될 수 있다고 보여져요.
이건 제가 아직 email 전송 라이브러리를 잘 몰라서 그런거라 재윤님 판단 하에 MimeMessage를 어디에 둘지 결정하시면 좋을 것 같아요~

Reply:
`MimeMessage` 가 'org.springframework.boot:spring-boot-starter-mail' 에 있는 거라서 인프라에 두었습니다.

추후 템플릿 작성과 메일 전송 부분은 분리하는게 좋아보입니다. 해당 부분은 학습을 해본 뒤 적용 해보도록 하겠습니다.

Reply:
따로 보고 있는 자료가 있어서 일단 `body` 부분은 그대로 두었습니다.